Content-Controls/Replace-picture-content-control-with-chart/.NET/Replace-picture-content-control-with-chart/Program.cs

```diff
@@ -0,0 +1,145 @@
+﻿using Syncfusion.DocIO.DLS;
+using Syncfusion.DocIO;
+using Syncfusion.DocIORenderer;
+using Syncfusion.OfficeChart;
+
+
+// Open the file as stream.
+using (FileStream docStream = new FileStream(Path.GetFullPath(@"../../../Data/PictureContentControl_InputDocument.docx"), FileMode.Open))
+{
+    // Load file stream into Word document.
+    using (WordDocument document = new WordDocument(docStream, Syncfusion.DocIO.FormatType.Automatic))
+    {
+        string[] propertyNames = { "ContentControlProperties.Title", "ContentControlProperties.Type" };
+        string[] propertyValues = { "Chart", "ContentControlType.Picture" };
+        // Find BlockContentControl by given properties
+        BlockContentControl contentControl = document.FindItemByProperty(EntityType.BlockContentControl, "ContentControlProperties.Title", "Chart") as BlockContentControl;
+
+        if (contentControl != null)
+        {
+            int index = contentControl.OwnerTextBody.ChildEntities.IndexOf(contentControl);
+
+            // Create a new paragraph to hold the chart image.
+            WParagraph paragraph = new WParagraph(document);
+
+            // Generate the chart and get it as an image stream.
+            Stream chartImage = GenerateChartAndGetChartAsImage();
+
+            // Create a new image instance and load the chart image.
+            WPicture picture = (WPicture)paragraph.AppendPicture(chartImage);
+
+            // Set picture dimensions.
+            picture.Height = 300;
+            picture.Width = 300;
+
+            // To replace Picture content Control insert Picture paragraph and remove the content control.
+            contentControl.OwnerTextBody.ChildEntities.Insert(index, paragraph);
+            contentControl.OwnerTextBody.ChildEntities.RemoveAt(index + 1);
+        }
+
+
+        // Create file stream for output document.
+        using (FileStream outputFileStream = new FileStream(Path.GetFullPath(@"../../../Data/ReplacedPictureContentControl.docx"), FileMode.Create, FileAccess.ReadWrite))
```

Comment:
Save the output file name as "Sample.docx"

Reply:
modified sir